utcaApp/src/udriver.h

```diff
@@ -23,29 +24,60 @@
 #include "enumerate.h"
 #include "pcie-single.h"
 
+struct ParamInit {
+    const char *name;
+    int &parameter;
+    std::optional<unsigned> number_of_channels = std::nullopt;
+    asynParamType type;
+    bool write_only = false;
+
+    ParamInit(const char *name, int &parameter, asynParamType type = asynParamInt32):
+        name(name),
+        parameter(parameter),
+        type(type)
+    {
+    }
+
+    ParamInit set_nc(unsigned new_number_of_channels)
+    {
+        number_of_channels = new_number_of_channels;
+        return *this;
+    }
+
+    ParamInit set_wo()
+    {
+        write_only = true;
+        return *this;
+    }
```

Comment:
The one I was thinking was something like this:

```diff
diff --git a/utcaApp/src/udriver.h b/utcaApp/src/udriver.h
index b005270..deea5b3 100644
--- a/utcaApp/src/udriver.h
+++ b/utcaApp/src/udriver.h
@@ -31,23 +31,18 @@ struct ParamInit {
     asynParamType type;
     bool write_only = false;
 
-    ParamInit(const char *name, int &parameter, asynParamType type = asynParamInt32):
+    ParamInit(const char *name, int &parameter, bool write_only = false, asynParamType type = asynParamInt32):
         name(name),
         parameter(parameter),
+        write_only(write_only),
         type(type)
     {
     }
 
-    ParamInit set_nc(unsigned new_number_of_channels)
+    ParamInit(const char *name, int &parameter, unsigned number_of_channels, bool write_only = false, asynParamType type = asynParamInt32):
+        ParamInit(name, parameter, write_only, type)
     {
-        number_of_channels = new_number_of_channels;
-        return *this;
-    }
-
-    ParamInit set_wo()
-    {
-        write_only = true;
-        return *this;
+        this->number_of_channels = number_of_channels;
     }
 };
```

But I didn't notice that would lead to ambiguous constructors due to `bool` and `unsigned`. That would also lead to a possibly less readable initialization nevertheless:

```diff
diff --git a/utcaApp/src/afc_timing.cpp b/utcaApp/src/afc_timing.cpp
index 1bcc841..9e637fa 100644
--- a/utcaApp/src/afc_timing.cpp
+++ b/utcaApp/src/afc_timing.cpp
@@ -43,10 +43,10 @@ class AFCTiming: public UDriver {
               {"STA_LOCKED_AFC_LATCH", p_afc_lock_latch},
               {"STA_LOCKED_RTM_LATCH", p_rtm_lock_latch},
               {"STA_LOCKED_GT0_LATCH", p_gt0_lock_latch},
-              ParamInit{"STA_LOCKED_RST_LATCH", p_lock_latch_rst}.set_wo(),
+              {"STA_LOCKED_RST_LATCH", p_lock_latch_rst, true},
           },
           {
-              ParamInit{"RFREQ_HI", p_rfreq_hi}.set_nc(2),
+              ParamInit{"RFREQ_HI", p_rfreq_hi, 2},
               ParamInit{"RFREQ_LO", p_rfreq_lo}.set_nc(2),
               ParamInit{"N1", p_n1}.set_nc(2),
               ParamInit{"HS_DIV", p_hs_div}.set_nc(2),
```

It seems to be a bad idea after all.